### PR TITLE
fix(hydra): store and use hydra context in a local variable

### PR DESCRIPTION
**Pull request by valentin-dassonville**
| Q             | A
| ------------- | ---
| Branch       | 4.0
| License       | MIT

### CODE REVIEW

**Comment by vincentchalamon** [comment]
@soyuka it seems to work on https://github.com/api-platform/demo/pull/472

**Comment by dunglas** [comment]
Can't we merge the Hydra context in the context we generate? So we avoid an extra HTTP request.

File: src/JsonLd/ContextBuilderInterface.php [hunk 1]
```diff
@@ -53,4 +52,905 @@ public function getResourceContext(string $resourceClass, int $referenceType = U
      * Gets the URI of the given resource context.
      */
     public function getResourceContextUri(string $resourceClass, int $referenceType = UrlGeneratorInterface::ABS_PATH): string;
+
+    public const HYDRA_CONTEXT = [
+        '@context' => [
```
Comment (dunglas) [inline src/JsonLd/ContextBuilderInterface.php:57]: Maybe could we store this in an external PHP file that we automatically generate from the upstream JSON (so we can regenerate the file if needed).

File: update-hydra-context.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\JsonLd;
+
+const HYDRA_CONTEXT = ".var_export($hydraContext, true).';';
+
+fwrite($file, $str);
+fclose($file);
```
Comment (dunglas) [inline update-hydra-context.php:43]:
```suggestion
file_put_contents('src/JsonLd/HydraContext.php', '<?php

/*
 * This is an autogenerated file, DO NOT MODIFY IT.
 * Run the update-hydra-contex.php script at the root of the project to refresh it.
 */

declare(strict_types=1);

namespace ApiPlatform\JsonLd;

const HYDRA_CONTEXT = '.var_export($hydraContext, true).';';
```

**Review by dunglas — APPROVED** [review]

**Comment by soyuka** [comment]
Thanks @valentin-dassonville !

**Pull request merged**